crates/cli/src/subcommands/publish.rs

```diff
@@ -303,22 +307,37 @@ async fn apply_pre_publish_if_needed(
     )
     .await?
     {
-        println!("{}", pre.migrate_plan);
-
-        if pre.break_clients
-            && !y_or_n(
-                break_clients_flag,
-                "The above changes will BREAK existing clients. Do you want to proceed?",
-            )?
-        {
-            println!("Aborting");
-            // Early exit: return an error or a special signal. Here we bail out by returning Err.
-            anyhow::bail!("Publishing aborted by user");
+        match pre {
+            PrePublishResult::ManualMigrate(manual) => {
+                if matches!(clear_database, ClearMode::OnConflict) {
+                    println!("{}", manual.reason);
+                    println!("Proceeding with database clear due to --delete-data=on-conflict.");
+                }
+                if matches!(clear_database, ClearMode::Never) {
+                    println!("{}", manual.reason);
+                    println!("Aborting publish due to required manual migration.");
+                    anyhow::bail!("Publishing aborted by user");
```

Comment:
is this a correct bail message? Sure the user is ultimately responsible, but isn't it really just that they _didn't_ choose to pass `--clear-data`? that feels like the user _not_ doing something.

Reply:
Oh, yes it is. Good catch. Will fix.